### PR TITLE
Upgrade eslint

### PR DESCRIPTION
**Pull request by doug-wade**
Clean up a bunch of eslint errors while I'm at it.

### CODE REVIEW

File: packages/react-server/core/logging/server.js [hunk 1]
```diff
@@ -1,163 +1,163 @@
...
-		timestamp : global.TIMESTAMP_REACT_SERVER_LOG_OUTPUT,
+var	winston	=	require('winston')
+,			common		=	require('./common')
+,			_	=	{
+	mapValues					:	require("lodash/object/mapValues"),
+	pick										:	require("lodash/object/pick"),
+	isPlainObject	:	require("lodash/lang/isPlainObject"),
+	isEmpty							:	require("lodash/lang/isEmpty"),
+	trimLeft						:	require("lodash/string/trimLeft"),
+	trunc									:	require("lodash/string/trunc"),
```
Comment (gigabo) [inline packages/react-server/core/logging/server.js]: No space between `:` and `require`?  😐

File: packages/react-server/core/logging/server.js [hunk 2]
```diff
@@ -1,163 +1,163 @@
...
+		level					:	config.baseLevel,
+		stream				:	process.stdout,
+		json						:	false,
+		timestamp	:	global.TIMESTAMP_REACT_SERVER_LOG_OUTPUT,
 	});
 
-	var logger = new (winston.Logger)({
-		transports: [
+	var	logger	=	new	(winston.Logger)({
+		transports:	[
```
Comment (gigabo) [inline packages/react-server/core/logging/server.js]: What?  Is github just rendering this weirdly?  Where did all the whitespace go?

File: packages/react-server/core/logging/server.js [hunk 3]
```diff
@@ -1,163 +1,163 @@
...
 
-var getLogger = common.makeGetLogger(makeLogger);
+var	getLogger	=	common.makeGetLogger(makeLogger);
 
-var colorizeName = function(opts){
+var	colorizeName	=	function(opts){
 
-	// Only colorize if we're attached to a terminal.
-	if (!global.COLORIZE_REACT_SERVER_LOG_OUTPUT) return opts.name;
+	//	Only	colorize	if	we're	attached	to	a	terminal.
```
Comment (gigabo) [inline packages/react-server/core/logging/server.js]: Uh...

File: packages/react-server/core/logging/server.js (hunk 1 above)
Comment (gigabo) [inline packages/react-server/core/logging/server.js]: Whoa, and now that I see this on the main PR page the tabling is all off... is it using _tabs_ for tabling?  That's generally not a good idea.  People set their tab stops at various positions.  Tabs work for _initial_ whitespace, but they don't work well for whitespace within a line.

File: packages/react-server/core/logging/server.js (hunk 2 above)
Comment (gigabo) [inline packages/react-server/core/logging/server.js]: Ah, yeah, I see now.   A bunch of spaces were just changed to tabs.

Let's not have tabs except at the beginning of lines.

Comment (doug-wade) [inline packages/react-server/core/logging/server.js]: Yeah, Atom's "convert spaces to tabs" function got a little overzealous...

File: packages/react-server/core/test/specRuntime/TransitionPage.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 var React = require("react"),  //eslint-disable-line no-unused-vars
-	Link = require("../../components/Link");
+	Link = require("../../components/Link"); //eslint-disable-line no-unused-vars
```
Comment (gigabo) [inline packages/react-server/core/test/specRuntime/TransitionPage.js]: `eslint-plugin-react` should make this unnecessary...

File: packages/react-server/core/components/RootElement.js [hunk 5]
```diff
@@ -49,7 +49,7 @@ class RootElement extends React.Component {
 			);
 
 			// Don't keep choking on it.  Just gut it.
-			return <div />;
+			return (<div></div>);
```
Comment (gigabo) [inline packages/react-server/core/components/RootElement.js]: What rules are forcing the parentheses and non-self-closing components?

File: packages/babel-preset-react-server/index.js [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
-    require('babel-plugin-transform-es2015-parameters'),
-    require('babel-plugin-transform-es2015-shorthand-properties'),
-    require('babel-plugin-transform-es2015-spread'),
-    require('babel-plugin-transform-es2015-template-literals'),
-    require('babel-plugin-transform-object-rest-spread'),
-  ],
-  presets: [
-    require('babel-preset-react'),
-  ],
+module.exports	=	{
```
Comment (gigabo) [inline packages/babel-preset-react-server/index.js]: I don't get this one.  Removing leading/trailing whitespace?

Comment (gigabo) [inline packages/babel-preset-react-server/index.js]: Oh, it's another set of mid-line tabs.  Weird that it renders differently in the PR's diff view vs the conversation view.

Comment (gigabo) [inline packages/babel-preset-react-server/index.js]: But I guess that's kind of the nature of mid-line tabs. ;)

**Comment by gigabo** [comment]
@doug-wade - This seems really close to being ready to merge.  Do want.  Ping.

File: packages/react-server/core/test/specRuntime/TransitionPage.js (hunk 4 above)
Comment (doug-wade) [inline packages/react-server/core/test/specRuntime/TransitionPage.js]: Turns out we were missing some rules we should've enabled: https://github.com/yannickcr/eslint-plugin-react#user-content-recommended-configuration

**Comment by gigabo** [comment]
@doug-wade - Looks like some conflicts have crept in here.

Looks like there are also still a few non-leading tabs.

**Comment by gigabo** [comment]
Dooouuuugggg.....

**Comment by gigabo** [comment]
Oh, hey... I upgraded eslint as part of #132.

@doug-wade - Anything you'd still like to pull in here beyond the simple upgrade?

**Pull request closed without merging**